src/clarity-angular/data/datagrid/providers/selection.ts

```diff
@@ -154,11 +167,33 @@ export class Selection {
     /**
      * Checks if an item is currently selected
      */
-    public isSelected(item: any): boolean {
-        if (this._selectionType === SelectionType.Single) {
-            return this.currentSingle === item;
-        } else if (this._selectionType === SelectionType.Multi) {
-            return this.current.indexOf(item) >= 0;
+    public isSelected(item: any, rowIndex?: number): boolean {
+        // Check if we are running with ngFor
+        if (!this.allReferences.length) {
+            const trackBy = this._items.trackBy;
+            if (this._selectionType === SelectionType.Single) {
+                // We use the rowIndex for trackBy as provided by the grid, but this won't really work in reality.
+                // Since each row is destroyed and recreated with pagination, the index will not reflect the true
+                // index of the item in the list (because it is not known to the grid in anyway), so this is
+                // broken but broken in a way that tries to use the closest approximation of index.
+                return (this.currentSingle) ? trackBy(rowIndex, this.currentSingle) === trackBy(rowIndex, item) : false;
+            } else if (this._selectionType === SelectionType.Multi) {
+                return this.current.findIndex((selected: any) => {
+                    // Same as the comment above with rowIndex
+                    return trackBy(rowIndex, selected) === trackBy(rowIndex, item);
+                }) >= 0;
+            }
+            return false;
+        }
+
+        // Have to lookup against the precalculated
+        const index = this.allReferences.findIndex((refItem: Reference) => {
+            return refItem.original === item;
+        });
+        if (this._selectionType === SelectionType.Single && this.allReferences[index]) {
+            return (this.currentSingle) ? this.currentSingle === this.allReferences[index].original : false;
```

Comment:
So _if_ we did need to use trackBy here, wouldn't it be the opposite? Rows always have the latest reference to the item, but `this.currentSingle` is the one that might be stale.